ff/serving/python_client/sdk/providers.py

```diff
@@ -0,0 +1,58 @@
+providers_to_register = []
+
+class Provider:
+    def __init__(self, name = str, provider_type = str, host = str, db = int, ssl = bool, client_certificate= str, bucket = str):
+        self.name = name
+        self.provider_type = provider_type
+        self.host = host
+        self.db = db
+        self.ssl = ssl
+        self.client_certificate = client_certificate
```

Comment:
Every provider will have different parameters. For example, it doesn't make sense for anything except S3 to have a bucket parameter.